Modix.Services/Giveaways/GiveawayService.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Discord;
+
+using Modix.Data.Models.Core;
+using Modix.Services.Core;
+
+namespace Modix.Services.Giveaways
+{
+    /// <summary>
+    /// Describes a service that provides operations for hosting giveaways.
+    /// </summary>
+    public interface IGiveawayService
+    {
+        /// <summary>
+        /// Determines the winners for the supplied giveaway message.
+        /// </summary>
+        /// <param name="message">The giveaway message that users reacted to.</param>
+        /// <param name="count">The number of winners to return.</param>
+        /// <returns>
+        /// A <see cref="ValueTask"/> that will complete when the operation is complete,
+        /// containing a result type that indicates whether the operation succeeded,
+        /// including the results if so or an error message if not.
+        /// </returns>
+        ValueTask<WinnersResult> GetWinnersAsync(IUserMessage message, int count);
```

Comment:
Per Microsoft:

```
Only if performance analysis proves it worthwhile should a ValueTask<TResult> be used instead of a Task<TResult>. 
```

I'm inclined to agree, although we could definitely examine where in the codebase `ValueTask` might be appropriate in the future.

Reply:
I've been using it on APIs that have a decent likelihood of just returning synchronously, though I suppose this method might not necessarily fall into that category